### ISSUE TITLE
get_thermal_name() sometimes chooses the wrong thermal scattering library

### ISSUE THREAD

**Issue by tjlaboss** [mit-crpg/openmc#935]
Some thermal scattering library names are close matches of the wrong libraries. For instance, "c_O_in_D2O" is considered a close match of "D_in_D2O" and "c_H_in_YH2" is considered a close match of "H_in_H2O". I've written a short test case below.

```python
# S(a,b) issue
#
# Demonstrates how the Python API can assign the wrong S(a,b) table
# when guessing at the proper name

import openmc

water = openmc.Material()
water.set_density("g/cc", 1.0)
water.add_nuclide("O16", 2.0/3, 'ao')
water.add_nuclide("H1", 0.9/3, 'ao')    # H in H2O
water.add_nuclide("H2", 0.1/3, 'ao')    # D in D2O
# Should correct this to "c_H_in_H2O"
water.add_s_alpha_beta("H_in_H2O")
# Should correct this to "c_D_in_D2O"
water.add_s_alpha_beta("D_in_D2O")
# It corrects these to the wrong libraries and warns the user.
print(water)
```

The output of which is

```
/home/travis/Research/openmc/openmc/data/thermal.py:92: UserWarning: Thermal scattering material "H_in_H2O" is not recognized. Assigning a name of c_H_in_YH2.
  'Assigning a name of {}.'.format(name, proper_name))
/home/travis/Research/openmc/openmc/material.py:654: UserWarning: OpenMC S(a,b) tables follow the GND naming convention. Table "H_in_H2O" is being renamed as "c_H_in_YH2".
  warnings.warn(msg)
/home/travis/Research/openmc/openmc/data/thermal.py:92: UserWarning: Thermal scattering material "D_in_D2O" is not recognized. Assigning a name of c_O_in_D2O.
  'Assigning a name of {}.'.format(name, proper_name))
/home/travis/Research/openmc/openmc/material.py:654: UserWarning: OpenMC S(a,b) tables follow the GND naming convention. Table "D_in_D2O" is being renamed as "c_O_in_D2O".
  warnings.warn(msg)
Material
	ID             =	1
	Name           =	
	Temperature    =	None
	Density        =	1.0 [g/cc]
	S(a,b) Tables  
	S(a,b)         =	('c_H_in_YH2', 1.0)
	S(a,b)         =	('c_O_in_D2O', 1.0)
	Nuclides       
	O16            =	0.6666666666666666 [ao]
	H1             =	0.3          [ao]
	H2             =	0.03333333333333333 [ao]
	Elements       
```

The mistake occurs in the function `get_thermal_name()` in [data/thermal.py](https://github.com/mit-crpg/openmc/blob/develop/openmc/data/thermal.py), which in turn calls `difflib.get_close_matches()`. One can change some of the `if` logic or possibly toy with the cutoff to make it smarter about catching these typos.

It's not a big issue, but might be fun for one of the new guys.

**Comment by paulromano** [comment]
I'm working on adding unit tests for our `openmc.data` package and this issue actually bit me so I went ahead and implemented a fix for it. I'll be submitting a PR soon.

**Comment by smharper** [comment]
Closed with #953
